haystack/schema.py

```diff
@@ -322,6 +323,32 @@ def run(self, *args: Any, **kwargs: Any):
         """
         pass
 
+    def _dispatch_run(self, **kwargs):
+        arguments = deepcopy(kwargs)
+        params = arguments.get("params", {})
+
+        run_signature_args = inspect.signature(self.run).parameters.keys()
+
+        run_params = {}
+        for key, value in params.items():
+            if key == self.name:  # targeted params for this node
+                if isinstance(value, dict):
+                    for _k, _v in value.items():
+                        if _k not in run_signature_args:
+                            raise Exception(f"Invalid parameter {_k} for the node {self.name}.")
+                run_params.update(**value)
+            elif not isinstance(value, dict) and key in run_signature_args:  # global params
+                run_params[key] = value
+
+        run_inputs = {}
+        for key, value in arguments.items():
+            if key in run_signature_args:
+                run_inputs[key] = value
+
+        output, stream = self.run(**run_inputs, **run_params)
+        output["params"] = params
```

Comment:
I think we'd need to append the unused `arguments` here to `output` (e.g. if we have `documents` but they are not used by the following node, only the one afterwards)